lib/SILOptimizer/Transforms/PerformanceInliner.cpp

```diff
@@ -351,6 +349,11 @@ bool SILPerformanceInliner::isProfitableToInline(FullApplySite AI,
   int BaseBenefit = RemovedCallBenefit;
 
   SubstitutionMap CalleeSubstMap;
+  if (IsGeneric) {
+    CalleeSubstMap = Callee->getGenericEnvironment()
+      ->getSubstitutionMap(AI.getSubstitutions());
```

Comment:
@slavapestov please construct this map only on demand. There is no need to always create it.